WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java

```diff
@@ -158,12 +156,12 @@
     /*
      * used when called from comment list
      */
-    static CommentDetailFragment newInstance(SiteModel site, CommentModel comment) {
+    static CommentDetailFragment newInstance(SiteModel site, CommentModel commentModel) {
         CommentDetailFragment fragment = new CommentDetailFragment();
         Bundle args = new Bundle();
         args.putInt(KEY_MODE, FROM_BLOG_COMMENT);
-        args.putSerializable(WordPress.SITE, site);
-        args.putSerializable(KEY_COMMENT, comment);
+        args.putLong(KEY_SITE_ID, site.getSiteId());
```

Comment:
Using the `siteId` won't work here. This is a bit tricky, but the WordPress app supports 2 kind of sites: WordPress.com sites and Self Hosted sites. The `siteId` is the id set by remote sites, and it's usable to distinguish WordPress.com sites, but can't be used for self hosted sites (different sites could have the same id).

Having two kind of ids (remote and local) is a PITA, this is the reason we were passing the full serialized models (`SiteModel` and `CommentModel`) between activities -> fragments, here in Comments but also in other parts of the app. This is not a good practice, but that allows us to abstract ids and only use them when there are needed (network layer).

Reply:
To fix this, you can either pass local ids (`site.getId()` int) or use the serialized site model.

Reply:
This makes sense. What do you think about only using local ids? It's a primary key so it should be unique and we only use it here to load the object from the DB. @nbradbury was concerned that storing full objects might lead to the "transaction too large" [issue](https://issuetracker.google.com/issues/37103380#comment77)

Reply:
Confirmed that it fails if you try to open the detail view on a self hosted site:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int org.wordpress.android.fluxc.model.SiteModel.getId()' on a null object reference
  at org.wordpress.android.fluxc.persistence.CommentSqlUtils.getCommentBySiteAndRemoteId(CommentSqlUtils.java:95)
  at org.wordpress.android.fluxc.store.CommentStore.getCommentBySiteAndRemoteId(CommentStore.java:201)
  at org.wordpress.android.ui.comments.CommentDetailFragment.setComment(CommentDetailFragment.java:408)
  at org.wordpress.android.ui.comments.CommentDetailFragment.onCreate(CommentDetailFragment.java:191)
  at android.app.Fragment.performCreate(Fragment.java:2031)
```

Reply:
Just to confirm, I'm now using local ID for site model and remote id for comments. Since we always request comments with a combination of site + comment remote ID, it should be safe. I tested using comment local ID is not safe and can change. What do you think?